maven/api-maven-archive/src/main/java/org/jboss/shrinkwrap/resolver/api/maven/archive/assembler/PomEquippedArchiveMavenAssembler.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2012, Red Hat Middleware LLC, and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.shrinkwrap.resolver.api.maven.archive.assembler;
+
+import org.jboss.shrinkwrap.api.Assignable;
+import org.jboss.shrinkwrap.resolver.api.maven.ScopeType;
+import org.jboss.shrinkwrap.resolver.api.maven.filter.MavenResolutionFilter;
+
+/**
+ * Instance of {@link ArchiveMavenAssembler} that has configuration from a POM file loaded
+ *
+ * @author <a href="mailto:mjobanek@redhat.com">Matous Jobanek</a>
+ * @author <a href="kpiwko@redhat.com">Karel Piwko</a>
+ */
+public interface PomEquippedArchiveMavenAssembler extends Assignable {
+
+    /**
+     * Based on metadata previously loaded from a Project Object Model file, adds content made during package phase to
+     * the archive. The content is compiled using dependencies with following scopes:
+     * {@link ScopeType#COMPILE}, {@link ScopeType#IMPORT}, {@link ScopeType#PROVIDED}, {@link ScopeType#RUNTIME},
+     * {@link ScopeType#SYSTEM} - there isn't used the scope {@link ScopeType#TEST} as it simulates Maven package build.
+     * <p>
+     * If you want to use all scopes, then use this method: {@code withBuildOutput(ScopeType.values())}
+     * </p>
+     *
+     * @return This modified {@link ArchiveMavenAssembler} instance with added content based on metadata previously loaded
+     * from a Project Object Model file.
+     */
+    PomEquippedArchiveMavenAssembler withBuildOutput();
+
+    /**
+     * Based on metadata previously loaded from a Project Object Model file, adds content (classes and resources) made
+     * during package phase to the archive.
+     * The content is compiled using dependencies with the set of scopes given by the parameter
+     * <p>
+     * If you want to use all scopes, then use: {@code withBuildOutput(ScopeType.values())}
+     * </p>
+     *
+     * @return This modified {@link ArchiveMavenAssembler} instance with added content based on metadata previously loaded
+     * from a Project Object Model file.
+     */
+    PomEquippedArchiveMavenAssembler withBuildOutput(ScopeType... scopes);
```

Comment:
Is at least one ScopeType required?  Maybe note the requirements for this in docs.  And if one *is* required, you can change signature to `withBuildOutput(ScopeType scope, ScopeType... scopes)` to compiler-force the user to put at least one in.